### PR TITLE
New: Switch from `ember-ajax` to `ember-fetch`

### PR DESCRIPTION
**Pull request by balajimanoharan**
New: Switch from `ember-ajax` to `ember-fetch`
- add new error utils for handling fetch errors

### CODE REVIEW

File: packages/data/addon/adapters/bard-facts.js [hunk 1]
```diff
@@ -325,17 +322,16 @@ export default EmberObject.extend({
...
-      },
-      beforeSend(xhr) {
-        xhr.setRequestHeader('clientid', clientId);
-        Object.keys(customHeaders).forEach(name => xhr.setRequestHeader(name, customHeaders[name]));
-      },
-      crossDomain: true,
-      data: query,
-      timeout: timeout
-    });
+    Object.keys(query).forEach(key => url.searchParams.append(key, query[key]));
```
Comment (jkusa) [inline packages/data/addon/adapters/bard-facts.js]:
```js
Object.entries(query).forEach(pair => url.searchParams.append(...pair));
```

File: packages/data/addon/adapters/bard-metadata.js [hunk 2]
```diff
@@ -61,21 +56,21 @@ export default EmberObject.extend({
...
       clientId = options.clientId || 'UI',
       timeout = options.timeout || 300000;
 
-    return this.get('ajax').request(url, {
-      xhrFields: {
-        withCredentials: true
-      },
-      beforeSend: function(xhr) {
-        xhr.setRequestHeader('clientid', clientId);
+    Object.keys(query).forEach(key => url.searchParams.append(key, query[key]));
```
Comment (jkusa) [inline packages/data/addon/adapters/bard-metadata.js]: Same as above

File: packages/data/addon/adapters/dimensions/bard.js [hunk 3]
```diff
@@ -173,17 +169,17 @@ export default EmberObject.extend({
       }
     }
 
-    return get(this, 'ajax').request(url, {
-      xhrFields: {
-        withCredentials: true
-      },
-      beforeSend: function(xhr) {
-        xhr.setRequestHeader('clientid', clientId);
+    Object.keys(filterQuery).forEach(key => url.searchParams.append(key, filterQuery[key]));
```
Comment (jkusa) [inline packages/data/addon/adapters/dimensions/bard.js]: Same

File: packages/data/addon/utils/errors.js [hunk 4]
```diff
@@ -0,0 +1,38 @@
...
+  return response;
+}
+
+/**
+ * @function IsJsonString
+ *
+ * @param {String} str
+ * @returns {Boolean}
+ */
+function IsJsonString(str) {
```
Comment (jkusa) [inline packages/data/addon/utils/errors.js]: `isJsonString`?

File: packages/data/addon/utils/errors.js [hunk 5]
```diff
@@ -0,0 +1,38 @@
...
+ *
+ * Handles fetch errors
+ * @param {Object} response
+ * @throws {Error} error object
+ * @returns {Object} response object
+ */
+
+export function handleErrors(response) {
+  if (!response.ok) {
+    const { _bodyText: error } = response;
```
Comment (jkusa) [inline packages/data/addon/utils/errors.js]: Can we do
```js
const error = reponse.text();
```

File: packages/data/addon/adapters/dimensions/bard.js [hunk 6]
```diff
@@ -132,7 +128,7 @@ export default EmberObject.extend({
   },
 
   /**
-   * @method find - Uses the url generated using the adapter to make an ajax request
+   * @method find - Uses the url generated using the adapter to make an fetch request
```
Comment (Alex-Aralis) [inline packages/data/addon/adapters/dimensions/bard.js]: make a fetch

File: packages/data/addon/adapters/bard-facts.js [hunk 7]
```diff
@@ -290,7 +287,7 @@ export default EmberObject.extend({
   },
 
   /**
-   * @method fetchDataForRequest - Uses the url generated using the adapter to make an ajax request
+   * @method fetchDataForRequest - Uses the url generated using the adapter to make an fetch request
```
Comment (Alex-Aralis) [inline packages/data/addon/adapters/bard-facts.js]: an -> a

**Comment by balajimanoharan** [comment]
a little more work needed to have consistent errors between packages. Will reopen when ready.

**Pull request closed without merging**